src/App.js

```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import { HashRouter as Router, Route } from "react-router-dom";
+import { Provider } from "react-redux";
+
+import store from "./store";
+import { getProducts } from "./storeReducers/productsReducer";
+import { checkSessionLogin } from "./storeReducers/userReducer";
+import { getCart } from "./storeReducers/cartReducer";
+
+import { Home } from "./components/Home";
+import Header from "./components/Header";
+import DetailProduct from "./components/DetailedProduct";
+import { CreateUserForm } from "./components/CreateUserForm";
+import CreateReview from "./components/CreateReview";
+import UserProfile from "./components/UserProfile";
+
+
+/* I think its reasonable to make this thing be the main provider of redux store
+ * And also the thing that routes to other places
+ * Lets see how this works!
+ */
+
+class App extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      loading: true
+    };
+  }
+
+  async componentDidMount() {
```

Comment:
I don't know that i've ever seen an `async componentDidMount` I'd probably recommend against it - also, this looks like you guys arent using thunks.

In short: We should never have asynchronous methods in a react component, we should mitigate side effects to be handled somewhere else. The `thunk` thinking is that we call a function that may or may not succeed - it can call the store when either event happens. Here - if we fail any of these awaits it directly affects rendering. This is not the react way - react is declarative and functional and ${name a buzzword} but what Im trying to say is keep it pure - side effects shouldnt affect our components.